lib/runtime-core/src/parse.rs

```diff
@@ -183,45 +196,54 @@ pub fn read_module<
                     ExternalKind::Global => ExportIndex::Global(GlobalIndex::new(index as usize)),
                 };
 
-                info.exports.insert(field.to_string(), export_index);
+                info.write()
+                    .unwrap()
+                    .exports
+                    .insert(field.to_string(), export_index);
             }
             ParserState::StartSectionEntry(start_index) => {
-                info.start_func = Some(FuncIndex::new(start_index as usize));
+                info.write().unwrap().start_func = Some(FuncIndex::new(start_index as usize));
             }
             ParserState::BeginFunctionBody { .. } => {
                 let id = func_count.wrapping_add(1);
                 func_count = id;
                 if func_count == 0 {
-                    info.namespace_table = namespace_builder.take().unwrap().finish();
-                    info.name_table = name_builder.take().unwrap().finish();
-                    mcg.feed_signatures(info.signatures.clone())
+                    info.write().unwrap().namespace_table =
+                        namespace_builder.take().expect("214").finish();
```

Comment:
Is this expect a debug?

Reply:
I'll add these to the todo list for cleanup.